Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -485,6 +500,30 @@ public class MapboxNavigationService: NSObject, NavigationService {
                   routerType: routerType,
                   customActivityType: customActivityType)
     }
+
+    init(indexedRouteResponse: IndexedRouteResponse,
+         customRoutingProvider: RoutingProvider?,
+         credentials: Credentials,
+         locationSource: NavigationLocationManager,
+         eventsManagerType: NavigationEventsManager.Type?,
+         simulating simulationMode: SimulationMode,
+         routerType: Router.Type?,
+         customActivityType: CLActivityType?,
+         simulatedLocationManagerFactory: SimulatedLocationManagerFactory,
```

Comment:
replace with `SimulatedLocationManager.Type?`, so it will be consistent with the way we already create `eventsManager`

Reply:
At least I don't think we need a separate property `private var simulatedLocationManagerFactory`, because we can pass it to `commonInit` directly

Reply:
At first didn't want to mark the initializer `public required` to be able to call it on the type.  
But you are right , it's better not to introduce the second approach for the same problem. Even if it lead to a new public constructor.

And I need to store `simulatedLocationSourceType` because `simulatedLocationSource` instance can be recreated more than once